sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/FileAPITests.groovy

```diff
@@ -124,12 +123,66 @@ class FileAPITests extends APISpec {
 
     def "Upload data error"() {
         when:
-        primaryFileClient.upload(defaultData.retain(), dataLength, 1, FileRangeWriteType.UPDATE)
+        primaryFileClient.upload(defaultData.retain(), dataLength, 1)
         then:
         def e = thrown(StorageErrorException)
         FileTestHelper.assertExceptionStatusCodeAndMessage(e, 404, StorageErrorCode.RESOURCE_NOT_FOUND)
     }
 
+    def "Upload and clear range" () {
+        given:
+        def fullInfoString = "please clear the range"
+        def fullInfoData = Unpooled.wrappedBuffer(fullInfoString.getBytes(StandardCharsets.UTF_8))
+        primaryFileClient.create(fullInfoString.length())
+        primaryFileClient.upload(fullInfoData.retain(), fullInfoString.length())
+        when:
+        primaryFileClient.clearRange(7)
+        def downloadResponse = primaryFileClient.downloadWithProperties()
+        then:
+        FileTestHelper.equalsAfterTrim(7, 0,
+            FluxUtil.collectBytesInByteBufStream(downloadResponse.value().body(), false).block())
+    }
+
+    def "Upload and clear range with args" () {
+        given:
+        def fullInfoString = "please clear the range"
+        def fullInfoData = Unpooled.wrappedBuffer(fullInfoString.getBytes(StandardCharsets.UTF_8))
+        primaryFileClient.create(fullInfoString.length())
+        primaryFileClient.upload(fullInfoData.retain(), fullInfoString.length())
+        when:
+        primaryFileClient.clearRange(7, 1)
+        def downloadResponse = primaryFileClient.downloadWithProperties()
```

Comment:
Instead of requiring a helper method to test range clearing why not download a range and assert that it is a zero'd out byte array?

Reply:
Done.